examples/keras_recipes/creating_tfrecords.py

```diff
@@ -0,0 +1,291 @@
+
+An interesting use case of TFRecords are TPUs because the hardware is multi-core, and its
+speed usually benefits from optimized I/O operations, besides that TPUs usually require
+data to be stored remotely (e.g. Google storage), and storing your data as TFRecords
+makes it easier to the load data without batch-downloading it.
+
+Performance with TFRecords can be further improved if later, you use it with the
+[tf.data](https://www.tensorflow.org/guide/data) API.
+
+Here we will learn how to convert data of different types (image, text, and numeric) into
+TFRecords.
+
+**Reference**
+
+- [TFRecord and tf.train.Example](https://www.tensorflow.org/tutorials/load_data/tfrecord)
+
+
+## Dependencies
+"""
+
+import os
+import json
+import pprint
+import tensorflow as tf
+import matplotlib.pyplot as plt
+
+"""
+## Downloading the data
+"""
+
+root_dir = "datasets"
+tfrecords_dir = "tfrecords"
+images_dir = os.path.join(root_dir, "val2017")
+annotations_dir = os.path.join(root_dir, "annotations")
+annotation_file = os.path.join(annotations_dir, "instances_val2017.json")
+images_url = "http://images.cocodataset.org/zips/val2017.zip"
+annotations_url = (
+    "http://images.cocodataset.org/annotations/annotations_trainval2017.zip"
+)
+
+# Download image files
+if not os.path.exists(images_dir):
+    image_zip = tf.keras.utils.get_file(
+        "images.zip", cache_dir=os.path.abspath("."), origin=images_url, extract=True,
+    )
+    os.remove(image_zip)
+
+# Download caption annotation files
+if not os.path.exists(annotations_dir):
+    annotation_zip = tf.keras.utils.get_file(
+        "captions.zip",
+        cache_dir=os.path.abspath("."),
+        origin=annotations_url,
+        extract=True,
+    )
+    os.remove(annotation_zip)
+
+print("Dataset is downloaded and extracted successfully.")
+
+with open(annotation_file, "r") as f:
+    annotations = json.load(f)["annotations"]
+
+print(f"Number of images: {len(annotations)}")
+
+"""
+## The COCO2017 dataset
+
+We will be using the [COCO2017](https://cocodataset.org/) dataset because it has many
+different types of features and will serve as a good example of how to encode different
+features as TFRecords.
+
+This dataset has two sets of features, images, and the instances meta-data.
+
+Images are a collection of ".jpg" files, and the meta-data is a ".json" file that
+according to the [official site](https://cocodataset.org/#format-data) contains these
+properties:
+```
+id: int,
+image_id: int,
+category_id: int,
+segmentation: RLE or [polygon], object segmentation mask
+bbox: [x,y,width,height], object bounding box coordinates
+area: float, area of the bounding box
+iscrowd: 0 or 1, is single object or a collection
+```
+
+### Looking at the annotations of a single data sample
+"""
+
+pprint.pprint(annotations[60])
+
+"""
+## Parameters
+
+`n_samples` is the number of data samples on each TFRecord file.
+
+`n_tfrecods` is total number of TFRecords that we will create.
+"""
+
+n_samples = 4096
```

Comment:
For consistency with other examples, use `num_` as the number prefix (everywhere).